src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.ManifestResources.cs

```diff
@@ -22,144 +21,118 @@ internal sealed partial class ExecutionEnvironmentImplementation : ExecutionEnvi
+                NativeParser entryParser;
+                while (!(entryParser = lookup.GetNext()).IsNull)
+                {
+                    if (entryParser.StringEquals(assemblyName))
                     {
-                        NativeReader reader;
-                        if (!TryGetNativeReaderForBlob(module, ReflectionMapBlob.BlobIdResourceIndex, out reader))
-                        {
-                            continue;
-                        }
-                        NativeParser indexParser = new NativeParser(reader, 0);
-                        NativeHashtable indexHashTable = new NativeHashtable(indexParser);
-
-                        var entryEnumerator = indexHashTable.EnumerateAllEntries();
-                        NativeParser entryParser;
-                        while (!(entryParser = entryEnumerator.GetNext()).IsNull)
-                        {
-                            string assemblyName = entryParser.GetString();
-                            string resourceName = entryParser.GetString();
-                            int resourceOffset = (int)entryParser.GetUnsigned();
-                            int resourceLength = (int)entryParser.GetUnsigned();
-
-                            ResourceInfo resourceInfo = new ResourceInfo(resourceName, resourceOffset, resourceLength, module);
-
-                            LowLevelList<ResourceInfo> assemblyResources;
-                            if (!dict.TryGetValue(assemblyName, out assemblyResources))
-                            {
-                                assemblyResources = new LowLevelList<ResourceInfo>();
-                                dict[assemblyName] = assemblyResources;
-                            }
-
-                            assemblyResources.Add(resourceInfo);
-                        }
+                        entryParser.SkipString(); // assemblyName
+                        yield return entryParser.GetString();
                     }
-
-                    s_extractedResourceDictionary = dict;
+                    else
+                    {
+                        entryParser.SkipString(); // assemblyName
+                        entryParser.SkipString(); // resourceName
+                    }
+                    entryParser.SkipInteger(); // offset
+                    entryParser.SkipInteger(); // length
                 }
-                return s_extractedResourceDictionary;
-            }
-        }
-
-        /// <summary>
-        /// This dictionary gets us from assembly + resource name to the offset of a resource
-        /// inside the resource data blob
-        ///
-        /// The dictionary's key is a Fusion-style assembly name.
-        /// The dictionary's value is a list of (resourcename,index) tuples.
-        /// </summary>
-        private static volatile LowLevelDictionary<string, LowLevelList<ResourceInfo>> s_extractedResourceDictionary;
-
-        private struct ResourceInfo
```

Comment:
BTW: It is ok to define small private helper structs like this one to pass things around so that the code is more readable. They cost nothing in NativeAOT and very little in regular CoreCLR.